hs_core/models.py

```diff
@@ -1523,7 +1523,8 @@ class Bags(models.Model):
     content_type = models.ForeignKey(ContentType)
 
     content_object = generic.GenericForeignKey('content_type', 'object_id')
-    bag = models.FileField(upload_to='bags', max_length=500, storage=IrodsStorage() if getattr(settings,'USE_IRODS', False) else DefaultStorage(), null=True) # actually never null
+    #bag = models.FileField(upload_to='bags', max_length=500, storage=IrodsStorage() if getattr(settings,'USE_IRODS', False) else DefaultStorage(), null=True) # actually never null
+    bag_url = models.URLField(null=True, blank=True)
```

Comment:
My main concern was that the absolute URL not be stored in the database, so it is good you are not storing that.  However, I don't think we should be storing even the relative URL.  The problem is, if we decide we need to edit urls.py in django_irods changing "download" to something else, then the database contains broken URLs that we then have to deal with some how.  Even these relative URLs should be generated dynamically at run time, for example using [reverse](https://docs.djangoproject.com/en/1.7/topics/http/urls/#reversing-namespaced-urls).

Also, I suggest changing django_irods/urls.py so that the:

    url(r'^download/$', 'django_irods.views.download'),

is instead:

    url(r'^download/(?P<path>[^/]+)$', 'django_irods.views.download'),

then when you reverse do something like:

    reverse('django_irods:django_irods.views.download', kwargs={'path': 'BAGNAME'})

Let me know if I can help with any of this.